src/run_info.c

```diff
@@ -344,4 +344,12 @@ int RAI_RunInfoBatchable(struct RAI_DagOp *op1, struct RAI_DagOp *op2) {
 
     return 1;
 }
-RAI_ModelRunCtx *RAI_GetModelRunCtx(RedisAI_RunInfo *rinfo) { return rinfo->dagOps[0]->mctx; }
+RAI_ModelRunCtx *RAI_GetAsModelRunCtx(RedisAI_RunInfo *rinfo, RAI_Error *err) {
+
+    RAI_DagOp *op = rinfo->dagOps[0];
+    RAI_SetError(err, RAI_GetErrorCode(op->err), RAI_GetError(op->err));
+    RAI_ModelRunCtx *mctx = op->mctx;
```

Comment:
what happens if user do this operation over a DAG which is not modelrun?
Please validate that `mctx` is actually model run context and return NULL in case it doesn't.

also, why duplicate the error?

Reply:
User should not call this in a DAGRUN OnFinish CB, but only in a MODELRUN. I can set this in the error msg. 
The error is not duplicated, we copy from the runInfo to the RAI_Error that the user sent and then we remove the runInfo (which contains the error in its first op, but not in the mctx).